spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binding/OutputBindingLifecycle.java

```diff
@@ -46,13 +47,13 @@ public void setApplicationContext(ApplicationContext applicationContext)
 	@Override
 	public void start() {
 		if (!running) {
-
+			ChannelBindingServiceProperties channelBindingServiceProperties = null;
 			// retrieve the ChannelBindingService lazily, avoiding early initialization
 			try {
 				ChannelBindingService channelBindingService = this.applicationContext
 						.getBean(ChannelBindingService.class);
-				Map<String, Bindable> bindables = this.applicationContext
-						.getBeansOfType(Bindable.class);
+				channelBindingServiceProperties = channelBindingService.getChannelBindingServiceProperties();
```

Comment:
We should inject the `ChannelBindingServiceProperties` bean directly, not retrieve it from the service.